operators/src/processing/point_in_polygon.rs

```diff
@@ -0,0 +1,775 @@
+    for InitializedPointInPolygonFilter
+{
+    fn query_processor(&self) -> Result<TypedVectorQueryProcessor> {
+        let point_processor = if let TypedVectorQueryProcessor::MultiPoint(point_processor) =
+            self.vector_sources[0].query_processor()?
+        {
+            point_processor
+        } else {
+            unreachable!("checked in `PointInPolygonFilter` constructor");
+        };
+
+        let polygon_processor = if let TypedVectorQueryProcessor::MultiPolygon(polygon_processor) =
+            self.vector_sources[1].query_processor()?
+        {
+            polygon_processor
+        } else {
+            unreachable!("checked in `PointInPolygonFilter` constructor");
+        };
+
+        Ok(TypedVectorQueryProcessor::MultiPoint(
+            PointInPolygonFilterProcessor::new(point_processor, polygon_processor).boxed(),
+        ))
+    }
+}
+
+pub struct PointInPolygonFilterProcessor {
+    points: Box<dyn VectorQueryProcessor<VectorType = MultiPointCollection>>,
+    polygons: Box<dyn VectorQueryProcessor<VectorType = MultiPolygonCollection>>,
+}
+
+impl PointInPolygonFilterProcessor {
+    pub fn new(
+        points: Box<dyn VectorQueryProcessor<VectorType = MultiPointCollection>>,
+        polygons: Box<dyn VectorQueryProcessor<VectorType = MultiPolygonCollection>>,
+    ) -> Self {
+        Self { points, polygons }
+    }
+
+    fn filter_points(
+        points: &MultiPointCollection,
+        polygons: &MultiPolygonCollection,
+        initial_filter: &BooleanArray,
+    ) -> Result<BooleanArray> {
+        let mut filter = Vec::with_capacity(points.len());
+
+        let tester = PointInPolygonTester::new(polygons);
+
+        let coordinates = points.coordinates();
+
+        for ((&coordinates_start_index, &coordinates_end_index), time_interval) in points
+            .multipoint_offsets()
+            .iter()
+            .tuple_windows()
+            .zip(points.time_intervals())
+        {
+            let (coordinates_start_index, coordinates_end_index) = (
+                coordinates_start_index as usize,
+                coordinates_end_index as usize,
+            );
+
+            let is_multi_point_in_polygon_collection = coordinates
+                [coordinates_start_index..coordinates_end_index]
+                .iter()
+                .any(|coordinate| tester.is_coordinate_in_collection(coordinate, time_interval));
+
+            filter.push(is_multi_point_in_polygon_collection);
+        }
+
+        arrow::compute::or(initial_filter, &filter.into()).map_err(Into::into)
+    }
+}
+
+impl VectorQueryProcessor for PointInPolygonFilterProcessor {
+    type VectorType = MultiPointCollection;
+
+    fn vector_query(
+        &self,
+        query: QueryRectangle,
+        ctx: QueryContext,
+    ) -> BoxStream<'_, Result<Self::VectorType>> {
+        // TODO: multi-threading
+
+        let filtered_stream = self
+            .points
```

Comment:
wouldn't it make more sense to loop over the polygons first? Otherwise you perform the precomputations for the polygons multiple times.

Reply:
No, because the output of the operator is `MultiPointCollection`. I would need to cache the point feature flags otherwise for the whole stream.

Reply:
so it's a trade-off between space and runtime? A bool array for all features in a stream does not strike me as infeasible to store. Or what do you mean with feature flags?

Reply:
Well, the total runtime is most-likely cheaper, since you do the polygon computations exactly once.
On the other hand, you have to store booleans for the whole stream, which is possible for sizes that are in our mind now, but imposes some limitation nevertheless.
Furthermore, you will get the first results later. For now, you can output the first point collection early, as soon as the first polyogn iteration is finished.
If you do it the other way around, you have to wait for len(polygons) * len(points) iterations for the first output, only in the last point iteration for the last set of polygons, you will get all the outputs.